rpc/server.go

```diff
@@ -490,7 +491,16 @@ func (conn *Conn) bindRequest(hdr *Header) (boundRequest, error) {
 
 // runRequest runs the given request and sends the reply.
 func (conn *Conn) runRequest(req boundRequest, arg reflect.Value, version int, observer Observer) {
+	// If the request causes a panic, ensure we log that before closing the connection.
+	defer func() {
+		if panicResult := recover(); panicResult != nil {
```

Comment:
Yeah - the agent just restarted "for no reason" and debug-log exited. Really hard to track down the issue. When I added this code, it was oh so easy to see the bug.